src/identity-providers/add/DescriptorSettings.tsx

```diff
@@ -0,0 +1,428 @@
+              <HelpItem
+                helpText={th("signatureAlgorithm")}
+                forLabel={t("signatureAlgorithm")}
+                forID="kc-signatureAlgorithm"
+              />
+            }
+            fieldId="kc-signatureAlgorithm"
+          >
+            <Controller
+              name="config.signatureAlgorithm"
+              defaultValue="RSA_SHA256"
+              control={control}
+              render={({ onChange, value }) => (
+                <Select
+                  toggleId="kc-signatureAlgorithm"
+                  onToggle={() =>
+                    setSignatureAlgorithmDropdownOpen(
+                      !signatureAlgorithmDropdownOpen
+                    )
+                  }
+                  isOpen={signatureAlgorithmDropdownOpen}
+                  onSelect={(_, value) => {
+                    onChange(value as string);
+                    setSignatureAlgorithmDropdownOpen(false);
+                  }}
+                  selections={value}
+                  variant={SelectVariant.single}
+                >
+                  <SelectOption key={0} value="RSA_SHA1" />
+                  <SelectOption key={1} value="RSA_SHA256" isPlaceholder />
+                  <SelectOption key={2} value="RSA_SHA256_MGF1" />
+                  <SelectOption key={3} value="RSA_SHA512" />
+                  <SelectOption key={4} value="RSA_SHA512_MGF1" />
+                  <SelectOption key={5} value="DSA_SHA1" />
+                </Select>
+              )}
+            ></Controller>
+          </FormGroup>
+          <FormGroup
+            label={t("samlSignatureKeyName")}
+            labelIcon={
+              <HelpItem
+                helpText={th("samlSignatureKeyName")}
+                forLabel={t("samlSignatureKeyName")}
+                forID="kc-samlSignatureKeyName"
+              />
+            }
+            fieldId="kc-samlSignatureKeyName"
+          >
+            <Controller
+              name="config.xmlSigKeyInfoKeyNameTransformer"
+              defaultValue="keyID-option"
+              control={control}
+              render={({ onChange, value }) => (
+                <Select
+                  toggleId="kc-samlSignatureKeyName"
+                  onToggle={() =>
+                    setSamlSignatureKeyNameDropdownOpen(
+                      !samlSignatureKeyNameDropdownOpen
+                    )
+                  }
+                  isOpen={samlSignatureKeyNameDropdownOpen}
+                  onSelect={(_, value) => {
+                    onChange(value as string);
+                    setSamlSignatureKeyNameDropdownOpen(false);
+                  }}
+                  selections={value}
+                  variant={SelectVariant.single}
+                >
+                  <SelectOption key={0} value="NONE" />
+                  <SelectOption key={1} value={t("keyID")} isPlaceholder />
+                  <SelectOption key={2} value={t("certSubject")} />
```

Comment:
I don't think it's needed to give these `SelectOption` a `key` prop, as they are statically populated.

```suggestion
                  <SelectOption value="NONE" />
                  <SelectOption value={t("keyID")} isPlaceholder />
                  <SelectOption value={t("certSubject")} />
```

Reply:
removed